src/main/java/org/opensearch/knn/index/codec/util/KNNCodecUtil.java

```diff
@@ -84,4 +90,36 @@ public static List<String> getEngineFiles(String extension, String fieldName, Se
             .collect(Collectors.toList());
         return engineFiles;
     }
+
+    public static String getEngineFileFromFieldInfo(FieldInfo field, SegmentInfo segmentInfo) {
+        if (!field.attributes().containsKey(KNN_FIELD)) {
+            return null;
+        }
+        // Only Native Engine put into indexPathMap
+        KNNEngine knnEngine = getNativeKNNEngine(field);
+        if (knnEngine == null) {
+            return null;
+        }
+        final List<String> engineFiles = KNNCodecUtil.getEngineFiles(knnEngine.getExtension(), field.name, segmentInfo);
+        final String vectorIndexFileName = engineFiles.get(0);
+        return vectorIndexFileName;
+    }
+
+    /**
+     * Get KNNEngine From FieldInfo
+     *
+     * @param field which field we need produce from engine
+     * @return if and only if Native Engine we return specific engine, else return null
+     */
+    private static KNNEngine getNativeKNNEngine(@NonNull FieldInfo field) {
+        final String modelId = field.getAttribute(MODEL_ID);
+        if (modelId != null) {
+            return null;
+        }
```

Comment:
This is redundant, FieldInforExtractor has the logic

Reply:
sure, removed in the PR.